### PR TITLE
feat: Add support for parsing CSV files with missing values

### PR DESCRIPTION
**Pull request by diegokolena**
This commit adds support for parsing CSV files with missing values. Four new CSV files were added to the test data directory, each containing different types of missing values. The  file contains missing values in the second column, the  file contains missing values in both columns, the  file contains missing values in the second column and JSON strings in the third and fourth columns, and the  file contains missing values in the first column.

Additionally, the  file was modified to include a new function  that attempts to parse empty strings as , numbers, and JSON strings into Python objects. This function is used to normalize the missing values in the CSV files.

This feature will improve the robustness of the CSV parsing functionality and allow for better handling of missing values in dataframes.

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/_utils/dataframes/transformers.py [hunk 1]
```diff
@@ -51,19 +54,97 @@ def _normalize_dicts(dict_list: List[Dict[str, Any]], sep: str = ".", max_level:
...
     all_keys = set().union(*flattened_dicts)
 
-    # Normalize all dictionaries to have all keys, filling missing ones with None
+    # Normalize all dictionaries to have all keys,
+    # filling missing ones with None and parsing existing ones
     for d in flattened_dicts:
         for key in all_keys:
-            if key not in d:
-                d[key] = None
+            d[key] = _try_parse(d.get(key, None))
```
Comment (yifanwu-kolena) [inline kolena/_utils/dataframes/transformers.py]: what's the rational to add parse at here? `_normalize_dicts` is only used by `json_normalize`.

File: kolena/_utils/dataframes/transformers.py [hunk 2]
```diff
@@ -60,10 +63,88 @@ def _normalize_dicts(dict_list: List[Dict[str, Any]], sep: str = ".", max_level:
     return flattened_dicts
 
 
+def _try_parse(value: Any) -> Any:
```
Comment (yifanwu-kolena) [inline kolena/_utils/dataframes/transformers.py:66]: i think we need to use this function in `_deserialize_dataobject_str` or somewhere else

File: kolena/_utils/dataframes/transformers.py [hunk 3]
```diff
@@ -60,10 +63,88 @@ def _normalize_dicts(dict_list: List[Dict[str, Any]], sep: str = ".", max_level:
...
+
+def drop_invalid_metadata(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Drops the invalid 'metadata' column from the DataFrame.
+    """
+    METADATA_FIELD = "metadata"
+    if METADATA_FIELD in df.columns:
+        if not isinstance(df[METADATA_FIELD].iloc[0], dict):
+            df.drop(columns=[METADATA_FIELD], inplace=True)
+    return df
```
Comment (yifanwu-kolena) [inline kolena/_utils/dataframes/transformers.py]: do we need these functions to be ported over for now?

File: kolena/_utils/dataframes/transformers.py (hunk 2 above)
Comment (consoull) [reply kolena/_utils/dataframes/transformers.py:66]: Ya, this function should be used in https://github.com/kolenaIO/kolena/blob/trunk/kolena/io.py#L61

**Comment by consoull** [comment]
Can we update the IO tests to catch regressions here

**Review by yifanwu-kolena — APPROVED** [review]

**Pull request merged**